src/ast/semantic_analyser.cpp

```diff
@@ -810,8 +811,7 @@ void SemanticAnalyser::visit(Call &call)
     for (auto &ap : *probe_->attach_points)
     {
       ProbeType type = probetype(ap->provider);
-      if (type != ProbeType::usdt && type != ProbeType::uretprobe &&
-          type != ProbeType::uprobe)
+      if (!check_available(call, type))
       {
         LOG(ERROR, call.loc, err_)
             << "uaddr can only be used with u(ret)probes and usdt probes";
```

Comment:
Feels like we now need to keep track of this in two places. We have to update check_available to reflect the probe types and then update the error message too. Can't we update check_available to output the err msg, the other checks do it as well.

Changing the signature to `check_available(Call &c, std::vector<probetype &> t)` might make that easier too.

Reply:
I changed so that the error messages are printed out when check_available() failed.
https://github.com/iovisor/bpftrace/pull/1646/commits/6fcf1a72913db4ea699ce8cbd7e7215cd9e70f77#diff-042d9701e67f79452452faac3c625c3ca83e7692abd3274956abe30d58ae0c39R459-R467